src/inverse-dynamics-balance-controller.cpp

```diff
@@ -362,39 +399,41 @@ void InverseDynamicsBalanceController::updateComOffset() {
 
 void InverseDynamicsBalanceController::removeRightFootContact(const double& transitionTime) {
   if (m_contactState == DOUBLE_SUPPORT) {
-    SEND_MSG("Remove right foot contact in " + toString(transitionTime) + " s", MSG_TYPE_INFO);
+    SEND_MSG("Remove right foot contact in " + toString(transitionTime) + " s", MSG_TYPE_ERROR);
     bool res = m_invDyn->removeRigidContact(m_contactRF->name(), transitionTime);
     if (!res) {
       const HQPData& hqpData = m_invDyn->computeProblemData(m_t, m_q_urdf, m_v_urdf);
       SEND_MSG("Error while remove right foot contact." + tsid::solvers::HQPDataToString(hqpData, false),
                MSG_TYPE_ERROR);
     }
     const double& w_feet = m_w_feetSIN.accessCopy();
-    m_invDyn->addMotionTask(*m_taskRF, w_feet, 1);
+    m_invDyn->addMotionTask(*m_taskRF, w_feet, 0);
     if (transitionTime > m_dt) {
       m_contactState = LEFT_SUPPORT_TRANSITION;
       m_contactTransitionTime = m_t + transitionTime;
-    } else
+    } else {
       m_contactState = LEFT_SUPPORT;
+    }
   }
 }
 
 void InverseDynamicsBalanceController::removeLeftFootContact(const double& transitionTime) {
   if (m_contactState == DOUBLE_SUPPORT) {
-    SEND_MSG("Remove left foot contact in " + toString(transitionTime) + " s", MSG_TYPE_INFO);
+    SEND_MSG("Remove left foot contact in " + toString(transitionTime) + " s", MSG_TYPE_ERROR);
     bool res = m_invDyn->removeRigidContact(m_contactLF->name(), transitionTime);
     if (!res) {
       const HQPData& hqpData = m_invDyn->computeProblemData(m_t, m_q_urdf, m_v_urdf);
       SEND_MSG("Error while remove right foot contact." + tsid::solvers::HQPDataToString(hqpData, false),
                MSG_TYPE_ERROR);
     }
     const double& w_feet = m_w_feetSIN.accessCopy();
-    m_invDyn->addMotionTask(*m_taskLF, w_feet, 1);
+    m_invDyn->addMotionTask(*m_taskLF, w_feet, 0);
```

Comment:
Beware that the more tasks you add as constraints, the more you risk getting an unfeasible problem

Reply:
Yes I agree but I achieve better results with this "priority" level in torque control.
However, when I use DCM control to achieve position control (integrate twice the acceleration), I have to put back these tasks to the cost (as for the CoM tasks -> one for the height and one for the rest).